src/openforms/formio/tests/validation/test_editgrid.py

```diff
@@ -510,3 +515,66 @@ class Serializer(serializers.Serializer):
             ]
         }
         self.assertEqual(data, expected)
+
+
+class EditGridNestedFilesTests(APITestCase):
+    """
+    Test the editgrid field specifically with nested files in repeating groups.
+
+    The submission needs to be in the context of the serializer's children (bound to the parent)
+    because backend validation of the file uses this.
+    """
+
+    def setUp(self):
+        self.addCleanup(clear_caches)
+        super().setUp()
+
+    @tag("dh-4656")
+    def test_nested_file_step_validate(self):
+        temporary_file_upload = TemporaryFileUploadFactory.create()
+        submission = temporary_file_upload.submission
+        file = SubmittedFileFactory.create(temporary_upload=temporary_file_upload)
+        form_step = FormStepFactory.create(
+            form=submission.form,
+            form_definition__configuration={
+                "components": [
+                    {
+                        "type": "editgrid",
+                        "key": "parent",
+                        "label": "Repeating group",
+                        "components": [
+                            {
+                                "key": "fileUpload1",
+                                "file": {
+                                    "name": "",
+                                    "type": [],
+                                    "allowedTypesLabels": [],
+                                },
+                                "type": "file",
+                                "label": "File Upload",
+                            },
+                        ],
+                    }
+                ]
+            },
+        )
+
+        session = self.client.session
+        ids = session.get("form-submissions", [])
+        ids += [str(submission.uuid)]
+        session["form-submissions"] = ids
+        session.save()
```

Comment:
tbh, I decided to add this kind of test since here https://github.com/open-formulieren/open-forms/blob/ea5282deac7704e1d6d6f57b13fcc7aa0ad7b558/src/openforms/formio/tests/validation/helpers.py#L22 we always have a submission in the context so this wouldn't really work in my situation (make sure that I do pass the submission from the parent)

Reply:
discussed on Slack :)